src/main/java/no/ndla/taxonomy/service/UrlResolverService.java

```diff
@@ -0,0 +1,109 @@
+        if (!results.isEmpty()) {
+            CachedUrlOldRig result = results.get(0);
+            List<String> allPaths = getAllPaths(result.getPublic_id());
+            if (result.getSubject_id() != null) {
+                String shortestPath = findshortestPathStartingWith(result.getSubject_id(), allPaths);
+                if (shortestPath != null) {
+                    return shortestPath;
+                } else {
+                    return getPrimaryPath(result.getPublic_id());
+                }
+            } else {
+                return getPrimaryPath(result.getPublic_id());
+            }
+        } else {
+            return null;
+        }
+    }
+
+    private String findshortestPathStartingWith(URI subject_id, List<String> allPaths) {
+        String subject = "/" + subject_id.toString().split("urn:")[1];
+        allPaths.sort(Comparator.comparing(String::length));
+        return allPaths.stream().filter(path -> path.startsWith(subject)).findFirst().orElse(null);
+    }
+
+    private List<String> getAllPaths(URI public_id) {
+        List<String> allpaths = jdbcTemplate.query("SELECT PATH, IS_PRIMARY FROM CACHED_URL WHERE PUBLIC_ID=?", setQueryParameters(Collections.singletonList(public_id.toString())),
+                (resultSet, rowNum) -> resultSet.getString("path")
+        );
+        return allpaths;
+    }
+
+    private String getPrimaryPath(URI public_id) {
+        List<String> primaryPaths = jdbcTemplate.query("SELECT PATH, IS_PRIMARY FROM CACHED_URL WHERE PUBLIC_ID=? AND IS_PRIMARY=TRUE", setQueryParameters(Collections.singletonList(public_id.toString())),
+                (resultSet, rowNum) -> resultSet.getString("path")
+        );
+        if (primaryPaths.isEmpty()) return null;
+        return primaryPaths.get(0);
+    }
+
+    private List<CachedUrlOldRig> getCachedUrlOldRig(String oldUrl) {
+        final String sql = "SELECT PUBLIC_ID, SUBJECT_ID FROM CACHED_URL_OLD_RIG WHERE OLD_URL=?";
+        final List<CachedUrlOldRig> query = jdbcTemplate.query(sql, setQueryParameters(asList(oldUrl)),
+                (resultSet, rowNum) -> new CachedUrlOldRig() {{
+                    setPublic_id(resultSet.getString("public_id"));
+                    if (resultSet.getString("subject_id") != null) {
+                        setSubject_id(resultSet.getString("subject_id"));
+                    }
+                }}
+        );
+        return query;
+    }
+
+    /**
+     * put old url into CACHED_URL_OLD_RIG
+     *
+     * @param oldUrl    url to put
+     * @param nodeId    nodeID to be associated with this URL
+     * @param subjectId subjectID to be associated with this URL (optional)
+     * @return true in order to be mockable "given" ugh!
+     * @throws Exception
+     */
+    public Boolean putPath(String oldUrl, URI nodeId, URI subjectId) throws Exception {
+        if (getAllPaths(nodeId).isEmpty())
+            throw new Exception();
```

Comment:
Burde det ikke være noen melding i denne Exception()?

Reply:
Har laget en ny checked exception som kastes av UrlResolverService.java - og enda en i controlleren som er HTTP - Dette vil også rette opp feilmeldingen nedenfor: _Element with url Node id not found in taxonomy for ndla.no/node/126644735684568 not found_, vil nå vises som _Node id not found in taxonomy for ndla.no/node/126644735684568_